pkg/kind/cluster_test.go

```diff
@@ -202,69 +195,3 @@ func TestGetConfigCustom(t *testing.T) {
 		assert.YAMLEq(t, string(expected), string(b))
 	}
 }
-
-// Mock provider for testing
-type mockProvider struct {
-	mock.Mock
-	IProvider
-}
-
-func (m *mockProvider) ListNodes(name string) ([]nodes.Node, error) {
-	args := m.Called(name)
-	return args.Get(0).([]nodes.Node), args.Error(1)
-}
-
-type mockRuntime struct {
-	mock.Mock
-}
-
-func (m *mockRuntime) ContainerWithPort(ctx context.Context, name string, port string) (bool, error) {
-	args := m.Called(ctx, name, port)
-	return args.Get(0).(bool), args.Error(1)
-}
-
-// Mock Docker client for testing
-type DockerClientMock struct {
-	client.APIClient
-	mock.Mock
-}
-
-func (m *DockerClientMock) ContainerList(ctx context.Context, listOptions types.ContainerListOptions) ([]types.Container, error) {
-	mockArgs := m.Called(ctx, listOptions)
-	return mockArgs.Get(0).([]types.Container), mockArgs.Error(1)
-}
-
-type NodeMock struct {
-	mock.Mock
-}
-
-func (n *NodeMock) Command(command string, args ...string) exec.Cmd {
-	argsMock := append([]string{command}, args...)
-	mockArgs := n.Called(argsMock)
-	return mockArgs.Get(0).(exec.Cmd)
-}
-
-func (n *NodeMock) String() string {
-	args := n.Called()
```

Comment:
I think we can leave mockProvider stuff  as they are. I know they are not used right now.